internal/services/voiceservices/voice_services_communications_gateway_resource.go

```diff
@@ -0,0 +1,618 @@
+					Codecs: []communicationsgateways.TeamsCodecs{
+						communicationsgateways.TeamsCodecs(model.Codecs),
+					},
+					E911Type:         model.E911Type,
+					Platforms:        expandCommunicationsPlatformModel(model.Platforms),
+					ServiceLocations: expandServiceRegionPropertiesModel(model.ServiceLocations),
+				},
+				Tags: &model.Tags,
+			}
+
+			if model.ApiBridge != "" {
+				var apiBridgeValue interface{}
+				err = json.Unmarshal([]byte(model.ApiBridge), &apiBridgeValue)
+				if err != nil {
+					return err
+				}
+				properties.Properties.ApiBridge = &apiBridgeValue
+			}
+
+			if model.EmergencyDialStrings != nil {
+				properties.Properties.EmergencyDialStrings = &model.EmergencyDialStrings
+			}
+
+			properties.Properties.OnPremMcpEnabled = &model.OnPremMcpEnabled
+
+			if model.TeamsVoicemailPilotNumber != "" {
+				properties.Properties.TeamsVoicemailPilotNumber = &model.TeamsVoicemailPilotNumber
+			}
+
+			if err := client.CreateOrUpdateThenPoll(ctx, id, *properties); err != nil {
+				return fmt.Errorf("creating %s: %+v", id, err)
+			}
+
+			metadata.SetID(id)
+
+			return nil
+		},
+	}
+}
+
+func (r CommunicationsGatewayResource) Update() sdk.ResourceFunc {
+	return sdk.ResourceFunc{
+		Timeout: 30 * time.Minute,
+		Func: func(ctx context.Context, metadata sdk.ResourceMetaData) error {
+			client := metadata.Client.VoiceServices.CommunicationsGatewaysClient
+
+			id, err := communicationsgateways.ParseCommunicationsGatewayID(metadata.ResourceData.Id())
+			if err != nil {
+				return err
+			}
+
+			var model CommunicationsGatewayModel
+			if err := metadata.Decode(&model); err != nil {
+				return fmt.Errorf("decoding: %+v", err)
+			}
+
+			resp, err := client.Get(ctx, *id)
+			if err != nil {
+				return fmt.Errorf("retrieving %s: %+v", *id, err)
+			}
+
+			properties := resp.Model
+			if properties == nil {
+				return fmt.Errorf("retrieving %s: properties was nil", id)
+			}
+
+			properties.SystemData = nil
+
+			if metadata.ResourceData.HasChange("codecs") {
+				properties.Properties.Codecs = []communicationsgateways.TeamsCodecs{
+					communicationsgateways.TeamsCodecs(model.Codecs),
+				}
+			}
+
+			if metadata.ResourceData.HasChange("e911_type") {
+				properties.Properties.E911Type = model.E911Type
+			}
+
+			if metadata.ResourceData.HasChange("platforms") {
+				properties.Properties.Platforms = expandCommunicationsPlatformModel(model.Platforms)
+			}
+
+			if metadata.ResourceData.HasChange("service_locations") {
+				properties.Properties.ServiceLocations = expandServiceRegionPropertiesModel(model.ServiceLocations)
+			}
+
+			if metadata.ResourceData.HasChange("api_bridge") {
+				if model.ApiBridge != "" {
+					var apiBridgeValue interface{}
+					err = json.Unmarshal([]byte(model.ApiBridge), &apiBridgeValue)
+					if err != nil {
+						return err
+					}
+					properties.Properties.ApiBridge = &apiBridgeValue
+				} else {
+					properties.Properties.ApiBridge = nil
+				}
```

Comment:
Since `ApiBridge` is `omitempty` this won't actually send anything if we set it to nil  
```suggestion
				var apiBridgeValue interface{}
				if model.ApiBridge != "" {
					log.Printf("[DEBUG] unmarshalling json for ApiBridge")
					if err = json.Unmarshal([]byte(model.ApiBridge), &apiBridgeValue); err != nil {
						return fmt.Errorf("unmarshalling value for ApiBridge: %+v", err)
					}
				}
				properties.Properties.ApiBridge = &apiBridgeValue
```